save-backend/Backend-API.md

```diff
@@ -0,0 +1,269 @@
+would like to evaluate and do few requests.\
+In this document will be represented the examples, which use `cURL`
+as instrument for working with SAVE Cloud API.
+
+To make the requests more universal, we will
+store major information, which is required by SAVE Cloud in requests, by
+into variables in **command prompt**.
+
+### Authorization configuration
+
+Each request, performed to API, require credentials, which could be configured like below, where
+
+* `SAVE_CLOUD_URL` - SAVE Cloud url and port
+* `SAVE_CLOUD_AUTH_SOURCE` where the identity is coming from, e.g. "github"
+* `SAVE_CLOUD_AUTH` - basic authorization header in format `source@username:token`.\
+   `source` is the same like `SAVE_CLOUD_AUTH_SOURCE`, and `token` is unique token,\
+    which could be created in personal settings in SAVE Cloud system.
+
+Just open up a command prompt and enter the following commands:
+
+```bash
+SAVE_CLOUD_URL=https://saveourtool.com:443
+
+SAVE_CLOUD_AUTH_SOURCE=github
+
+SAVE_CLOUD_AUTH='Basic Z2l0aHViQHVzZXJuYW1lOnRva2Vu'
+```
+
+### Configuration of information about evaluated tool
+
+To specify information about project, that you would like to evaluate,
+it's necessary to create the fields below in command promt. Only the `organizationName`,
+`projectName` and `gitUrl` are required, if your project is public:
+
+
+```bash
+# Required
+organizationName=Huawei
+
+# Required
+projectName='"save"'
+
+# Required
+gitUrl='"https://github.com/analysis-dev/save-cli"'
+
+# Optional
+sdkName='"openjdk"'
+sdkVersion='"11"'
+
+# Optional
+# Git username (generally, should be provided, only if tested project is private)
+gitUserName=null
+
+# Optional
+# Git password (generally, should be provided, only if tested project is private)
+gitPassword=null
+```
+
+**Note**: The `organizationName` here is intentionally don't contain quotes, since this value
+will be used in requests url, while other in requests bodies.
+
+There is two modes for execution, that could be performed: Git and Standard.\
+Each of them require own configuration.
+
+#### Configuration for Git Mode
+<details>
+  <summary>Expand</summary>
+
+Only the `testRootPath`, which represents 
+the relative path to the root directory with tests in your repository is required for execution.
+
+```bash
+# Required
+testRootPath='"examples/kotlin-diktat"'
+
+# Optional
+branch='"origin/feature/testing_for_cloud"'
+
+# Optional
+commitHash=null
+```
+
+</details>
+
+#### Configuration for Standard Mode
+
+For Standard Mode you need provide at least the list of test suites,
+which should be involved in testing
+
+<details>
+  <summary>Expand</summary>
+
+```bash
+# Required
+testSuites='["Directory: Chapter 1", "Directory: Chapter2"]'
+
+# Optional
+execCmd=null
+
+# Optional
+# Batch size controls how many files will be processed at the same time.
+batchSize=null
+
+```
+</details>
+
+### API for execution
+
+#### Upload additional files into SAVE Cloud storage
```

Comment:
Uploading is the most important part of API for CI/CD. You run  a build and upload binary - it should be clear and easy